docs/contributor/adr/0010-geoblocking-draft.md

```diff
@@ -0,0 +1,387 @@
+|                    |                                                         |                                                         |
+| Caching service    | Extensibility (may cache more things in future)         | Complex - another component to take care                |
+|                    | Works well with restart / upgrade / scaling             |                                                         |
+|                    | No downtime if geoblocking service doesn't work         |                                                         |
+|                    | Less load on geoblocking service                        |                                                         |
+|                    |                                                         |                                                         |
+| Config map         | Still simple                                            | Configmap capacity allows max ~50000 IP ranges          |
+|                    | Works well with restart / upgrade / scaling             | IP list integrity not guaranteed                        |
+|                    | No downtime if geoblocking service doesn't work         |                                                         |
+|                    | Less load on geoblocking service                        |                                                         |
+|                    |                                                         |                                                         |
+| Ephemeral volume   | Still simple                                            | Every container causes load on geoblocking service      |
+|                    | No downtime if geoblocking service doesn't work         | Doesn't help in case of restart / upgrade / scaling     |
+|                    |                                                         |                                                         |
+| Persistent volume  | No downtime if geoblocking service doesn't work         | Depends on the cloud infrastructure (ReadWriteMany)     |
+|                    | Less load on geoblocking service                        |                                                         |
+|                    | Works well with restart / upgrade / scaling             |                                                         |
+
+Decision: Let's use a Config Map as a fallback and cache for IP range allow/block list. However, let's split it technically from the Config Map that contains IP ranges provided by the end-user, so:
+- the Config Map containing custom IP range allow/block list is configured by the user, it becomes a contract
+- the Config Map containing IP range allow/block list downloaded from the SAP internal service is a Kyma internal thing and the implementation may change at any time, so no other module should use it
+
+Consequence: Config map capacity may be exceeded in future, which may require immediate attention.
```

Comment:
A consequence might also be that we want to monitor the size of the config map and get notified once we are reaching the threshold.

Reply:
Yes, I was also thinking about it. And it can be even done outside of ip-auth because everybody gets the same list of policies anyway. So we may be checking it even 'manually' in our local machines once per some time.

Reply:
I added an additional sentence saying that someone (we) should monitor the size of the policy list.